src/main.c

```diff
@@ -443,1273 +451,1609 @@ typedef struct {
+                          ImGuiInputTextFlags flags) {
+  return se_input_file_callback(label, new_path, types, NULL, flags);
 }
 void se_mkdir(char *path) {
-    char *sep = strrchr(path, '/');
-    if(sep != NULL) {
-        *sep = 0;
-        se_mkdir(path);
-        *sep = '/';
-    }
-    if(mkdir(path, 0777) && errno != EEXIST)
-        printf("error while trying to create '%s'\n%m\n", path); 
+  char *sep = strrchr(path, '/');
+  if (sep != NULL) {
+    *sep = 0;
+    se_mkdir(path);
+    *sep = '/';
+  }
+  if (mkdir(path, 0777) && errno != EEXIST)
+    printf("error while trying to create '%s'\n%m\n", path);
 }
 FILE *se_fopen_mkdir(const char *fpath, char *mode) {
-    char *path = strdup(fpath);
-    char *sep = strrchr(path, '/');
-    char *sep2 = strrchr(path, '\\');
-    if(sep2&&(!sep||sep2<sep))sep=sep2;
-    if(sep) { 
-        char *path0 = strdup(path);
-        path0[ sep - path ] = 0;
-        se_mkdir(path0);
-        free(path0);
-    }
-    free(path);
-    return fopen(fpath,mode);
-}
-void se_copy_file(const char * original_path, const char* copy_path){
-
-  FILE* source = fopen(original_path, "rb");
-  FILE* dest  = se_fopen_mkdir(copy_path, "wb");
-  if (source == NULL ||dest==NULL) {
+  char *path = strdup(fpath);
+  char *sep = strrchr(path, '/');
+  char *sep2 = strrchr(path, '\\');
+  if (sep2 && (!sep || sep2 < sep))
+    sep = sep2;
+  if (sep) {
+    char *path0 = strdup(path);
+    path0[sep - path] = 0;
+    se_mkdir(path0);
+    free(path0);
+  }
+  free(path);
+  return fopen(fpath, mode);
+}
+void se_copy_file(const char *original_path, const char *copy_path) {
+
+  FILE *source = fopen(original_path, "rb");
+  FILE *dest = se_fopen_mkdir(copy_path, "wb");
+  if (source == NULL || dest == NULL) {
     perror("Error opening source or dest file for copy\n");
-    if(source)fclose(source);
-    if(dest)fclose(dest);
-    return; 
+    if (source)
+      fclose(source);
+    if (dest)
+      fclose(dest);
+    return;
   }
   char buffer[1024];
-  size_t read = 0;;
+  size_t read = 0;
+  ;
 
   while ((read = fread(buffer, 1, sizeof(buffer), source)) > 0) {
-      fwrite(buffer, 1, read, dest);
+    fwrite(buffer, 1, read, dest);
   }
   fclose(source);
   fclose(dest);
 }
-void se_bios_file_open_fn(const char* dir){
-  //Make use of the fact that the accept function is called before the output path is updated.
-  char * se_bios_file_open_tmp_path = gui_state.file_browser.output_path;
-  if(strncmp(dir,se_bios_file_open_tmp_path,SB_FILE_PATH_SIZE)!=0){
-    if(sb_file_exists(se_bios_file_open_tmp_path)||strncmp(dir,"",SB_FILE_PATH_SIZE)==0)remove(se_bios_file_open_tmp_path);
-    if(strncmp(dir,"",SB_FILE_PATH_SIZE)!=0)se_copy_file(dir,se_bios_file_open_tmp_path);
-  }
-  emu_state.run_mode=SB_MODE_RESET;
-}
-
-const char* se_keycode_to_string(int keycode){
-  switch(keycode){
-    default:           return "Unknown";
-    case SAPP_KEYCODE_SPACE:         return "SPACE";
```

Comment:
It needs to not break up these case labels and the returns onto separate lines. 
They are laid out like this to allow block edits.